geps/gep-1619.md

```diff
@@ -157,49 +281,401 @@ Generally, the implementation API programs the dataplane API; however these two
+services or routes. It is defined as a [Direct Policy Attachment](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#direct-policy-attachment)
+without defaults or overrides, applied to the targeted service, HTTPRoute, or GRPCRoute.
+
+Attaching the `SessionPersistencePolicy` metaresource to a service will be a core support level feature, while attaching
+it to a route will be considered extended. Implementations must support services for conformance, while routes are
+considered optional. This distinction arises because most existing implementations primarily support attaching to a
+service, while attaching to a route is less common and involves greater complexities (see [API Attachment Point: Route vs. Service](#api-attachment-point-route-vs-service)
+for more details).
+
+#### API Attachment Point: Route vs. Service
+
+The `SessionPersistencePolicy` metaresource can target both a service and route object. The core functionality of
+attaching it to either is generally identical. The distinction lies in where the configuration gets propagated: when
+attached to a route, it can define session persistence for multiple services, whereas attaching it to a service defines
+it for a single service.
+
+Edge cases will arise when providing supporting both routes and services. For guidance on addressing conflicting
+attachments, please consult the [Expected API Behavior](#expected-api-behavior) section, which outlines API use cases.
+
+Whether attaching to a route or a service, it's important to consider the relationship the cookie has with the route.
+The cookie's `Path` attribute defines the URL path that must exist in order for the client to send the `cookie` header.
+Configuring the `Path` attribute may lead to potential issues with multiple routes pointing to the same service. For
+additional guidance on how implementors should handle the `Path` attribute, please refer to [Cookie Attributes](#cookie-attributes).
+
+#### Traffic Splitting
+
+In scenarios involving traffic splitting, session persistence impacts load balancing done after routing, regardless if
+it is attached to a service or a route. When a persistent session is established and traffic splitting is configured
+across services, the persistence to a single backend should be maintained across services. Consequently, a persistent
+session takes precedence over traffic split weights when selecting a backend after route matching. It's important to
+note that session persistence does not impact the process of route matching. See [Expected API Behavior](#expected-api-behavior)
+for example use cases on traffic splitting.
+
+#### Cookie Attributes
+
+While the API is intended to be generic, as described in [API Granularity](#api-granularity), a majority of
+implementations will employ session persistence through cookies. Therefore, let's explore the possibilities of cookie
+configuration for these APIs.
+
```

Comment:
@costinm I think "clients should not expect a specific name" could be highly contentious. Note that the client (whether gRPC or regular HTTP one) needs to copy the cookie value from the first response to subsequent requests so they need to know the cookie name. Either each implementation has a pre-baked name or this spec allows specifying a name and I prefer the latter option.

Reply:
CC @va7ish might be interested in providing input here.

Reply:
Added multiple cookie statement to `Session Initiation Guidelines` as it felt like it fit better there.

Added specific name statement in this section, but let's discuss. I thought it was standard protocol for clients to copy all cookies.

Reply:
Yes, AFAIK common practice is to copy all cookies ( with the rules defined by the standard on origin, path, secure, etc). But I'm not aware of any standard or only one cookie is copied. 

As shown in the 'prior art' - many load balancers have hardcoded names for the cookie - google, cloudfare, aws - and would not be practical for a client to guess.

Reply:
Some applications might require more tight control of the cookie used and some clients (which are application aware) can leverage that to provide addition application-specific optimizations. I think it's fairly important to allow the API users to control cookie attributes (name, ttl, path) when the underlying infra supports it. This could be achieved through an implementation-specific metaresource policy though.

Reply:
> I think it's fairly important to allow the API users to control cookie attributes (name, ttl, path) when the underlying infra supports it

I think the rules defined so far for Path under 'Cookie Attributes' would work. Name should preferably be configurable.  TTL is probably based on the value of `SessionPersistencePolicySpec. DurationSeconds`, Is that correct?

Reply:
This topic of (granular control of cookie attribute) might be an issue we should discuss in our next meeting, I added it to the agenda.

> TTL is probably based on the value of SessionPersistencePolicySpec. DurationSeconds, Is that correct?

Also mentioned in https://github.com/kubernetes-sigs/gateway-api/pull/2159/files#r1297887820. It could be just Cookie TTL, yes, but it's not limited to only the TTL cookie attribute and has the ability to be more complex than just Cookie TTL.

Reply:
Our [meeting](https://docs.google.com/document/d/1DYYyxM-O_JXuLPyWijabUbcl1cRBSFpX-OFi6eSijs8/edit#heading=h.gb5npubb0ftd)'s outcome was that we converged on allowing configuration of the `Name` attribute, as long as it was abstracted from the cookie implementation.

I added `SessionName` as a new field. Please review.

I also added/changed `AbsoluteTimeoutSeconds` and `IdleTimeoutSeconds` because I felt `DurationSeconds` was a little to ambiguous when considering it with the idle timeout, but please review and let me know if you disagree.